src/KlarnaOnsiteMessaging.php

```diff
@@ -133,7 +133,8 @@ public function enqueue_scripts() {
 		global $post;
 
 		$has_shortcode = ( ! empty( $post ) && has_shortcode( $post->post_content, 'onsite_messaging' ) );
-		if ( ! ( is_product() || is_cart() || ! $has_shortcode ) ) {
+		// NOTE! The order of these expressions is important. Always begin by checking for shortcode as it can be placed virtually anywhere.
+		if ( ! ( $has_shortcode || is_product() || is_cart() ) ) {
```

Comment:
This is not the same check as before.

Before you did:
`if ( ! ( is_product() || is_cart() || ! $has_shortcode ) )`
And now:
`if ( ! ( $has_shortcode || is_product() || is_cart() ) )`

So the Shortcode check has changed from `( ! ( ! $has_shortcode ) )` to `( ! ( $has_shortcode ) )`
Meaning that before if shortcode was false it would return false, now it returns true. And if it was true it would return true, and now returns false. Is that correct with how it should be?

Reply:
This is a mistake on my part. The `!` before `$has_shortcode` should not have been there in the first place. I've removed the comment as it is misleading.